build_tools/cmake/iree_benchmark_suite.cmake

```diff
@@ -284,23 +291,41 @@ function(iree_benchmark_suite)
         COMMENT "Generating ${_TOOL_FILE}"
       )
 
-      set(_TOOLFILE_GEN_TARGET_NAME_LIST "iree-generate-benchmark-toolfile")
-      list(APPEND _TOOLFILE_GEN_TARGET_NAME_LIST ${_COMMON_NAME_SEGMENTS})
-      list(JOIN _TOOLFILE_GEN_TARGET_NAME_LIST "__" _TOOLFILE_GEN_TARGET_NAME)
+      set(_TOOLFILE_GEN_TARGET_NAME
+        "${PACKAGE_NAME}_iree-generate-benchmark-toolfile__${_RUN_SPEC_TARGET_SUFFIX}")
       add_custom_target("${_TOOLFILE_GEN_TARGET_NAME}"
         DEPENDS "${_TOOL_FILE}"
       )
 
       # Generate a flagfile containing command-line options used to compile the
       # generated artifacts.
-      set(_COMPOPT_FILE "${_RUN_SPEC_DIR}/compilation_flagfile")
-      string(REPLACE ";" "\n" IREE_BENCHMARK_COMPILATION_FLAGS "${_TRANSLATION_ARGS}")
-      configure_file(
-        ${PROJECT_SOURCE_DIR}/build_tools/cmake/benchmark_compilation_flagfile.in
-        ${_COMPOPT_FILE})
+      set(_COMPILATION_FLAGFILE "${_RUN_SPEC_DIR}/compilation_flagfile")
+      # Generate the flagfile with python command. We can't use "file" because
+      # it can't be part of a target's dependency and generated lazily.
```

Comment:
Maybe also note that echo doesn't work with newlines

Reply:
Done.